packages/hooks/src/useRegister.ts

```diff
@@ -0,0 +1,175 @@
+import { SMART_INVOICE_FACTORY_ABI } from '@smartinvoicexyz/constants';
+import { logDebug, logError } from '@smartinvoicexyz/utils';
+import _ from 'lodash';
+import { useCallback, useMemo } from 'react';
+import { UseFormReturn } from 'react-hook-form';
+import { encodeAbiParameters, Hex, parseUnits, stringToHex } from 'viem';
+import { useSimulateContract, useWriteContract } from 'wagmi';
+
+import { SimulateContractErrorType, WriteContractErrorType } from './types';
+
+const REQUIRES_VERIFICATION = true;
+
+interface UseRegister {
+  escrowForm: UseFormReturn;
+  netwrokConfig: {
+    resolver: Hex;
+    tokenAddress: Hex;
+    wrappedNativeToken: Hex;
+    tokenDecimals: number;
+    factoryAddress: Hex;
+  };
+  onSuccess: (invoiceId: string | undefined) => Promise<void>;
+  enabled?: boolean;
+  details?: `0x${string}` | null;
+}
+
+export const useRegister = ({
```

Comment:
`useRegister` is essentially doing the same as `useInvoiceCreate` which is an existing hook, is it at all possible to reuse the existing code?